### ISSUE TITLE
[1.3.0-RC1] .pom files of dependencies in the local .m2 store get deleted while resolving dependencies

### ISSUE THREAD

**Issue by felher** [sbt/sbt#4709]
- [x] I've read the [CONTRIBUTING](https://github.com/sbt/sbt/blob/develop/CONTRIBUTING.md) guidelines

## steps
I attached a zip containing three things:
- a folder with a mini project, configured to show the problem
- a folder containing sample dependencies
- a docker file setting up a minimal, fresh system to show the problem (of course, you don't have to use the docker file. You can do everything locally, too)

I'm sorry the file is 17MB big, but I wanted to use the real dependencies I used and within in the folder were a few other depedencies.

To test it with a docker file, do the following in the root of demo folder after unzipping:
```
docker build -t felher/sbtrep:latest . # build the test system image
docker run --rm -ti felher/sbtrep:latest /bin/bash # start into the test system
ll /root/.m2/repository/de/uniwuerzburg/informatik/is/juriparser/juriparser/1.0/juriparser-1.0.pom #make sure the pom file is still there
cd /tmp/miniproject/ # change to the miniproject folder
sbt compile # start sbt and compile it. This takes a while, being a fresh install and all
ll /root/.m2/repository/de/uniwuerzburg/informatik/is/juriparser/juriparser/1.0/juriparser-1.0.pom  # now the pom file is gone.
```
You can also try it without using docker. Just unzip the file, copy the dependency folder to `~/.m2/repository/de/` and change into the miniproject folder and call `sbt compile` there.

## problem
After failing resolving the dependency, the .pom file got deleted.

## expectation
The dependency should have been found and the pom should not have been deleted.

## notes

sbt version: *1.3.0-RC1*. I also tried 1.0.1 and 1.2.8. Neither of them show this problem.
[sbtrep.zip](https://github.com/sbt/sbt/files/3195158/sbtrep.zip)

**Comment by eed3si9n** [comment]
@felher Thanks for the detailed reproduction.

/cc @alexarchambault

**Comment by eed3si9n** [comment]
I reported this to upstream (https://github.com/coursier/coursier/issues/1178) since I can reproduce this using sbt 1.2.8 with `addSbtPlugin("io.get-coursier" % "sbt-coursier" % "1.1.0-M13-4")`.

**Comment by eed3si9n** [comment]
Fixed by #4845

**Comment by felher** [comment]
Great, thanks! :+1:
